client.go

```diff
@@ -110,6 +119,80 @@ func chopContentType(contentType string) string {
 	return strings.Split(contentType, ";")[0]
 }
 
+// Subscribe function subscribes for events form the StreamClient TopicName after the given offset. An offset of zero should be
+// provided to read from the beginning. The provided EventHandler function will be called for each value.
+// To deal with errors while reading messages, an error handler function should also be provided.
+//
+// The function returns a context.CancelFunc which may be called for cancelling the subscription.
+func (lc *StreamClient) Subscribe(ctx context.Context, group string, offset uint64, f EventHandler, e EventErrHandler) (context.CancelFunc, error) {
+	subContext, cancel := context.WithCancel(ctx)
+	request := liiklus.SubscribeRequest{
+		Topic:                lc.TopicName,
+		Group:                group,
+		AutoOffsetReset:      liiklus.SubscribeRequest_EARLIEST,
+	}
+	subscribedClient, err := lc.client.Subscribe(subContext, &request)
+	if err != nil {
+		return cancel, err
+	}
+
+	subscribeReply, err := subscribedClient.Recv()
+	if err != nil {
+		return cancel, err
+	}
+
+	receiveRequest := liiklus.ReceiveRequest{
+		Assignment:           subscribeReply.GetAssignment(),
+		LastKnownOffset:      offset,
+	}
+	receiveClient, err := lc.client.Receive(subContext, &receiveRequest)
+	if err != nil {
+		return cancel, err
+	}
+
+	go func() {
+		for {
+			select {
+			case <- subContext.Done():
+				e(cancel, errors.New("context cancelled"))
```

Comment:
Is it always because the context was canceled that Done() is closed? Could be eg a timeout on the parent context, no?

Reply:
fair point. How about `context terminated`